### PR TITLE
StackDataset batch loading

### PR DESCRIPTION
**Pull request by stsouko**
data loading optimized

### CODE REVIEW

File: torch/utils/data/dataset.py [hunk 1]
```diff
@@ -254,6 +254,31 @@ def __getitem__(self, index):
             return {k: dataset[index] for k, dataset in self.datasets.items()}
         return tuple(dataset[index] for dataset in self.datasets)
 
+    def __getitems__(self, indices: List) -> List[T_stack]:
```
Comment (vadimkantorov) [inline torch/utils/data/dataset.py:257]: are the datasets in StackDataset supposed to have equal number of items?

Comment (stsouko) [reply torch/utils/data/dataset.py:257]: datasets should be the same by size, yes. good point.

Comment (stsouko) [reply torch/utils/data/dataset.py:257]: name stack has given as opposite to concatenate.
the behavior similar to torch.cat and torch.stack. 1st do flat tensor, 2nd creates new dim.

**Comment by stsouko** [comment]
@ejguan, could you please review the pr?
